File: from-0301-to-0400/0349-intersection-of-two-arrays/solution.go
func intersection(nums1 []int, nums2 []int) []int {
    imap1 := make(map[int]bool)
    for _, num := range nums1{
        imap1[num] = true
    }
    imap2 := make(map[int]bool)
    for _, num := range nums2{
        imap2[num] = true
    }
    result := []int{}
    for num, _ := range imap1 {
        if imap2[num] {
            result = append(result, num)
        }
    }
    return result
}

File: from-0301-to-0400/0349-intersection-of-two-arrays/solution.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        c1 = Counter(nums1)
        c2 = Counter(nums2)
        
        ans = []
        for k in c1.keys():
            if k in c2:
                ans.append(k)
                
        return ans
    